### PR TITLE
created objects to represent values returned by a cypher query

### PR DESCRIPTION
**Pull request by scferrada**

### CODE REVIEW

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/data/RecordEntry.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+    /**
+     * Gets the name of the column
+     * @return a CypherVar
+     */
+    CypherVar getName();
+
+    /**
+     * Gets the value of the column in this entry
+     * @return a Value
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/data/RecordEntry.java:19]: Also here, these `@return` parts do not add anything. You can remove them.

(Nonetheless, I appreciate that you are adding these JavaDoc comments!)

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/data/TableRecord.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+ *
+ * Each element of the record (ai: vi), is called an entry.
+ */
+public interface TableRecord {
+
+    /**
+     * Gets the collection of entries of the row/record.
+     * @return a Collection of RecordEntry
+     */
+    Collection<RecordEntry> getRecords();
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/data/TableRecord.java:18]: Also here, `Iterable<Record>` is more suitable as the return type.

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/data/impl/LPGEdge.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
+package se.liu.ida.hefquin.engine.wrappers.lpgwrapper.data.impl;
+
+import se.liu.ida.hefquin.engine.wrappers.lpgwrapper.data.PropertyMap;
+
+public class LPGEdge {
+
+    protected final String id;
+    protected final String label;
+    protected final PropertyMap properties;
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/data/impl/LPGEdge.java:9]: Should there be getter methods for these?

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/data/impl/LPGEdgeValue.java [hunk 4]
```diff
@@ -0,0 +1,10 @@
+package se.liu.ida.hefquin.engine.wrappers.lpgwrapper.data.impl;
+
+public class LPGEdgeValue {
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/data/impl/LPGEdgeValue.java:3]: I assume you wanted this to be:
```suggestion
public class LPGEdgeValue implements Value {
```

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/data/impl/LPGEdgeValue.java [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+    protected final LPGEdge edge;
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/data/impl/LPGEdgeValue.java:5]: Also here, what about a getter method for this member variable?

**Review by hartig — CHANGES_REQUESTED** [review]
There are a few things to be changed.

**Comment by scferrada** [comment]
I added a generic `get` method on the `Value` interface, I'm not really sold on it, since it must be generic enough to return whatever the concrete class is holding.

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/data/TableRecord.java [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+ * and vi are "values". A value can be a literal, a node, an edge, a list or a map.
+ *
+ * Each element of the record (ai: vi), is called an entry.
+ */
+public interface TableRecord {
+
+    /**
+     * Gets the collection of entries of the row/record.
+     */
+    Collection<RecordEntry> getRecordEntries();
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/data/TableRecord.java:17]: This should also be changed to `Iterable<RecordEntry>`

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/data/impl/ListValue.java [hunk 7]
```diff
@@ -0,0 +1,19 @@
...
+    protected final List<Object> values;
+
+    public ListValue(final List<Object> values) {
+        this.values = values;
+    }
+
+    @Override
+    public Object get() {
+        return values;
+    }
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/data/impl/ListValue.java:18]: same here

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/data/impl/LiteralValue.java [hunk 8]
```diff
@@ -0,0 +1,17 @@
...
+    protected final Object internalValue;
+
+    public LiteralValue(final Object internalValue) {
+        this.internalValue = internalValue;
+    }
+
+    @Override
+    public Object get() {
+        return internalValue;
+    }
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/data/impl/LiteralValue.java:16]:
```suggestion
    public Object getValue() {
        return internalValue;
    }
```

File: src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/data/impl/MapValue.java [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+    protected final Map<String, Object> values;
+
+    public MapValue(final Map<String, Object> values) {
+        this.values = values;
+    }
+
+    @Override
+    public Object get() {
+        return values;
+    }
```
Comment (hartig) [inline src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/data/impl/MapValue.java:18]: same here

**Review by hartig — CHANGES_REQUESTED** [review]

**Comment by hartig** [comment]
Right, such a generic getter method does not make sense. See the detailed comments in my review.

**Review by hartig — APPROVED** [review]

**Comment by hartig** [comment]
Looks good now. I will merge ...

**Comment by hartig** [comment]
Now with this PR merged, you may get back to PR #116 and you can create another PR with code for parsing a Cypher result string into a list of `TableRecord` objects.

**Pull request merged**